viewer/src/index.ts

```diff
@@ -1,16 +1,17 @@
-import { Parser } from "./parser";
+import { ComponentModelParser } from "./componentModelParser";
 import { DrawComponent } from "./drawComponent";
 
 const component = {
   model: document.currentScript.getAttribute("model"),
   graphical: document.currentScript.getAttribute("graphical")
 };
-const parser = new Parser(component);
-parser.parse((err, p) => {
-  if (err) {
+const parser = new ComponentModelParser(component);
+parser.parse()
+  .then(() => {
```

Comment:
Not a logical way to use the parser. It seems weird to put the parser as a parameter of drawComponent.
I think a better solution is: parser.parse().then((data) => ... drawComponent.draw(data...))
@maurelio1234 what do you think?